flatgfa/src/pool.rs

```diff
@@ -22,6 +22,23 @@ impl<T> Hash for Id<T> {
     }
 }
 
+impl<T> Add<u32> for Id<T> {
+    type Output = Self;
+
+    #[inline]
+    fn add(self, rhs: u32) -> Self::Output {
+        Self(self.0 + rhs, PhantomData)
```

Comment:
I'd probably write this as:

```
(self.index() + rhs).into()
```

…which has the benefit of reusing our existing code for `From`.

This probably doesn't matter for this PR, but I'm not 100% sure it is a good idea to support `Add` and `Sub` on `Id`s, as opposed to creating an explicit method like `Id::shift(n)`. The point is that `id + 5` or `id - 3` might look less scary than they should: because they are implicit, they might fool programmers into thinking that something innocuous is happening, whereas doing "index math" like this is actually somewhat advanced. So an explicit method like `shift` might be a better match for forcing programmers to tread carefully when they want to do this kind of index math.